fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/service/message/MessageFactory.java

```diff
@@ -33,10 +37,17 @@
 import org.apache.fineract.infrastructure.event.external.service.message.domain.MessageIdempotencyKey;
 import org.apache.fineract.infrastructure.event.external.service.message.domain.MessageSource;
 import org.apache.fineract.infrastructure.event.external.service.message.domain.MessageType;
+import org.apache.fineract.infrastructure.event.external.service.support.ByteBufferConverter;
+import org.springframework.beans.factory.InitializingBean;
 import org.springframework.stereotype.Component;
 
+@Slf4j
 @Component
-public class MessageFactory {
+@RequiredArgsConstructor
+public class MessageFactory implements InitializingBean {
+
+    private final ByteBufferConverter byteBufferConverter;
+    private static final String sourceUUID = UUID.randomUUID().toString();
```

Comment:
Naming convention. Should be named as SOURCE_UUID according to the Java standards.